test/dev/ui/spinner_test.rb

```diff
@@ -56,23 +85,24 @@ def printable(str)
         str.gsub(/\x1b\[[\d;]+\w/, '')
       end
 
+      COLOR_CODE_REGEX = /\x1b\[[\d;]+m/
       def match_lines(out, *patterns)
         # newline, or cursor-down
         lines = out.split(/\n|\x1b\[\d*B/)
 
         # Assert all lines are matched
-        lines.each do |l|
-          # strip ANSI colour code stuff
-          line = l.dup.gsub(/\x1b\[[\d;]+m/, '')
-          assert patterns.any? { |p| line.match(p) }, "Nothing matched the line #{line}"
+        lines.each_with_index do |l, index|
+          line = l.dup.gsub(COLOR_CODE_REGEX, '')
```

Comment:
there's a method for this hidden in dev-ui: https://github.com/Shopify/dev-ui/blob/master/lib/dev/ui/ansi.rb#L14-L16